airflow/hooks/subprocess.py

```diff
@@ -80,6 +85,13 @@ def pre_exec():
             self.sub_process.wait()
 
             self.log.info('Command exited with return code %s', self.sub_process.returncode)
+            return line
+
+        if cwd is None:
+            with TemporaryDirectory(prefix='airflowtmp') as tmp_dir:
+                line = __run_subprocess(tmp_dir)
+        else:
+            line = __run_subprocess(cwd)
```

Comment:
This can be more easily implemented with `contextlib.ExitStack`:

```python
with contextlib.ExitStack() as stack:
    if cwd is None:
        cwd = stack.enter_context(TemporaryDirectory(prefix='airflowtmp'))
    # Now you don’t need the __run_subprocess wrapper.
```

Reply:
Thanks for sharing this with me.
I've change it.